app/src/main/java/org/mozilla/focus/fragment/DownloadDialogFragment.java

```diff
@@ -0,0 +1,118 @@
+
+        String fileName = URLUtil.guessFileName(
+                download.getUrl(), download.getContentDisposition(), download.getMimeType());
+
+        args.putString("fileName", fileName);
+        args.putParcelable("download", download);
+        frag.setArguments(args);
+        return frag;
+    }
+
+    public AlertDialog onCreateDialog(Bundle bundle) {
+        final String fileName = getArguments().getString("fileName");
+        final Download pendingDownload = getArguments().getParcelable("download");
+
+        final AlertDialog.Builder builder = new AlertDialog.Builder(getContext(), R.style.DialogStyle);
+        builder.setCancelable(true);
+        builder.setTitle(getString(R.string.download_dialog_title));
+
+        final LayoutInflater inflater = getActivity().getLayoutInflater();
+        final View dialogView = inflater.inflate(R.layout.download_dialog, null);
+        builder.setView(dialogView);
+
+        final ImageView downloadDialogIcon = (ImageView) dialogView.findViewById(R.id.download_dialog_icon);
+        final TextView downloadDialogMessage = (TextView) dialogView.findViewById(R.id.download_dialog_file_name);
+        final Button downloadDialogCancelButton = (Button) dialogView.findViewById(R.id.download_dialog_cancel);
+        final Button downloadDialogDownloadButton = (Button) dialogView.findViewById(R.id.download_dialog_download);
+        final TextView downloadDialogWarningMessage = (TextView) dialogView.findViewById(R.id.download_dialog_warning);
+
+        downloadDialogIcon.setImageResource(R.drawable.ic_insert_drive_file_white_24px);
+
+        final String actionCancel = getString(R.string.download_dialog_action_cancel);
+        downloadDialogCancelButton.setText(actionCancel);
+
+        downloadDialogMessage.setText(fileName);
+
+        final String actionDownload = getString(R.string.download_dialog_action_download);
+        downloadDialogDownloadButton.setText(actionDownload);
+
+        downloadDialogWarningMessage.setText(getSpannedTextFromHtml(R.string.download_dialog_warning, R.string.app_name));
+
+        final AlertDialog alert = builder.create();
+
+        downloadDialogCancelButton.setOnClickListener(
+                new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        sendButtonClicked(pendingDownload, false);
+                        dismiss();
+                    }
+                });
+
+        downloadDialogDownloadButton.setOnClickListener(
+                new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        sendButtonClicked(pendingDownload, true);
+                        TelemetryWrapper.downloadDialogEvent();
+                        dismiss();
+                    }
+                });
+
+        return alert;
+    }
+
+    public interface DownloadDialogListener {
```

Comment:
nit: by convention, inner interfaces & classes should be at the bottom (or some cases, at the top!) of a class